### PR TITLE
Add and refactor e2e tests around edit enquiry

### PR DESCRIPTION
**Pull request by ian-leggett**
## Description
This is the final E2E tests around the edit enquiry journey. I have refactored previous work to be more DRY and added the final bits of code to finish the testing.

### CODE REVIEW

File: cypress/e2e_tests/support/commands.js [hunk 1]
```diff
@@ -1 +1,22 @@
-Cypress.Commands.add('findDetailsSection', number => cy.get('main').find('div > div').eq(number).find('h3'))
+Cypress.Commands.add('findDetailsSection', number => {
+  return number === 0
+    ? cy.get('main').find('div > div').eq(number).find('h2')
+    : cy.get('main').find('div > div').eq(number).find('h3')
+})
```
Comment (peterhudec) [inline cypress/e2e_tests/support/commands.js:5]: You could move the condition to the last `.find()`:
```js
Cypress.Commands.add('findDetailsSection', number =>
  cy.get('main')
    .find('div > div')
    .eq(number)
    .find(number === 0 ? 'h2' : 'h3')
)
```

File: cypress/e2e_tests/support/commands.js [hunk 2]
```diff
@@ -1 +1,22 @@
...
+const populateField = (type, name, value) => {
+  const fieldTypes = {
+    select: () => cy.get(`select[name=${name}]`).select(value),
+    textarea: () => cy.get(`textarea[name=${name}]`).clear().type(value),
+    default: () => cy.get(`input[name=${name}]`).clear().type(value),
+  }
+  return typeof fieldTypes[type] !== 'function'
+    ? fieldTypes['default']()
+    : fieldTypes[type]()
+}
```
Comment (peterhudec) [inline cypress/e2e_tests/support/commands.js:16]: I also tend to use objects for this kind of logic, just to realize that a switch statement is probably more readable:
```js
switch (type) {
  case 'select':
    return cy.get(`select[name=${name}]`).select(value)
  case 'textarea':
    return cy.get(`textarea[name=${name}]`).clear().type(value)
  default:
    return cy.get(`input[name=${name}]`).clear().type(value)
}
```

File: cypress/e2e_tests/support/commands.js (hunk 1 above)
Comment (ian-leggett) [reply cypress/e2e_tests/support/commands.js]: Thanks much better 👍

File: cypress/e2e_tests/support/commands.js (hunk 2 above)
Comment (ian-leggett) [reply cypress/e2e_tests/support/commands.js]: I agree, in this example its much clearer as a switch statement

**Pull request merged**